### PR TITLE
Category implementation

### PR DESCRIPTION
**Pull request by ponomvrenko**

### CODE REVIEW

**Review by DevIgork — APPROVED** [review]
lgtm

File: src/main/java/com/example/onlinebookstore/model/Category.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
```
Comment (d1sam) [inline src/main/java/com/example/onlinebookstore/model/Category.java:27]:
```suggestion
    @Column
    private String description;
```

File: src/main/java/com/example/onlinebookstore/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
```
Comment (d1sam) [inline src/main/java/com/example/onlinebookstore/model/Category.java:15]:
```suggestion
@Data
@NoArgsConstructor
```

File: src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        "Category with ID: " + id + ", unfortunately was not be found."
+                )
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryResponseDto updateById(Long id, CategoryRequestDto requestDto) {
+        Optional<Category> existingCategory = categoryRepository.findById(id);
+        if (existingCategory.isEmpty()) {
```
Comment (d1sam) [inline src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java:49]: let's better use `.orElseThrow`

File: src/main/resources/db/changelog/changes/09-create-categories-table.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        - createTable:
+            tableName: categories
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/09-create-categories-table.yaml]:
```suggestion
```
you have primary key, that's why nullable constraint is unnecessary

**Review by d1sam — CHANGES_REQUESTED** [review]
looks good, but some changes should be done :)

**Review by d1sam — APPROVED** [review]
well done!

**Pull request merged**